zipkin-ui/test/component_ui/traceSummary.test.js

```diff
@@ -540,3 +543,21 @@ describe('totalServiceTime', () => {
     totalServiceTime([time1, time2, time3, undefinedDuration]).should.equal(6000);
   });
 });
+
+describe('traceDuration', () => {
+  it('should return root span duration of synchronous trace', () => {
+    const rootSpan = httpTrace.find(span => !span.parentId);
+    traceDuration(httpTrace).should.equal(rootSpan.duration);
+    traceDuration(SPAN_V1.convertTrace(httpTrace)).should.equal(rootSpan.duration);
+  });
+
+  it('should return correct duration for a skewed async trace', () => {
```

Comment:
implicit todo here is to figure out how best to re-jig the data, and not use magic numbers in tests